backend/server/app.py

```
@@ -201,6 +205,256 @@ def convert_decimal_to_float(data):
+                "dte_grid": result.interpolation_grid[1].tolist() if result.interpolation_grid[1] is not None else None
+            }
+        }
+        
+        return JSONResponse(content=response_data)
+        
+    except HTTPException as http_exc:
+        logging.error(f"HTTP error in calibration: {http_exc.detail}")
+        raise http_exc
+    except Exception as e:
+        logging.error(f"Error in surface calibration: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Calibration failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)